test-framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java

```diff
@@ -132,16 +166,34 @@ public void sendRequest(DiscoveryNode node, long requestId, String action, Trans
     /**
      * Adds a rule that will cause matching operations to throw ConnectTransportExceptions
      */
-    public void addFailToSendNoConnectRule(DiscoveryNode node, final String... blockedActions) {
-        addFailToSendNoConnectRule(node, new HashSet<>(Arrays.asList(blockedActions)));
+    public void addFailToSendNoConnectRule(TransportService transportService, final String... blockedActions) {
+        for (TransportAddress transportAddress : extractTransportAddresses(transportService)) {
```

Comment:
I think the extraction here is redundant no - just pass the transport service...